include/secp256k1.h

```diff
@@ -577,9 +577,11 @@ SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_ec_pubkey_create(
 
 /** Negates a private key in place.
  *
- *  Returns: 1 always
+ *  Returns: 1 if seckey was successfully negated and 0 otherwise
  *  Args:   ctx:        pointer to a context object
- *  In/Out: seckey:     pointer to the 32-byte private key to be negated (cannot be NULL)
+ *  In/Out: seckey:     pointer to the 32-byte private key to be negated. The private key
+ *                      interpreted as an integer (most significant byte first) must be less than
+ *                      the curve order. (cannot be NULL)
```

Comment:
less than the curve order and not be 0?

(Actually we should just document what a valid private key is, e.g., here https://github.com/bitcoin-core/secp256k1/blob/master/include/secp256k1.h#L552)

Reply:
It won't fail for 0 though. I think ideally we document what's a valid secret key is (as you said) in the doc of `seckey_verify` and then just refer to that instead of having to restate it over and over. I can rewrite this PR to do that.